cdragontoolbox/hashes.py

```diff
@@ -495,15 +508,31 @@ def substitute_numbers(self, nmax=100, digits=None):
         paths = self.known.values()
         super()._substitute_numbers(paths, nmax, digits)
 
+    def check_basename_prefixes(self, prefixes=None):
+        """Checks a provided list of prefixes for all basenames.
+        If no list is provided, a default one will be used"""
+        values = set()
+        if prefixes is None:
+            prefixes = {'2x_', '2x_sd_', '4x_', '4x_sd_', 'sd_'}
```

Comment:
Could this be passed as the default param value instead?

Reply:
You mean like `` def check_basename_prefixes(self, prefixes={'2x_', '2x_sd_', '4x_', '4x_sd_', 'sd_'}):``?

Reply:
It's usually better to avoid using mutable objects as default parameters, because the same reference will be used each time.
For instance:
```py
def f(x=[]):
    x.append(1)
    return x
a = f()
b = f()
assert a is b
assert a == b == [1, 1]
```

In our case, `prefixes` is not modified by the function, so it would work. But it's easy to forget when updating the code and bugs could be introduced later on.

So IMHO Morilli's approach is the good one.